README.md

```diff
@@ -1387,36 +1361,32 @@ import org.folio.rest.tools.utils.TenantLoading;
 
 @Validate
 @Override
-public void postTenant(TenantAttributes tenantAttributes, Map<String, String> headers,
-    Handler<AsyncResult<Response>> handler, Context context) {
-  super.postTenant(tenantAttributes, headers, res -> {
-    if (res.failed()) {
-      handler.handle(res);
-      return;
-    }
+Future<Void> loadData(TenantAttributes attributes, String tenantId, Map<String, String> headers,
+  Context verxContext) {
+  Future<Void> future = super.loadData(attributes, tenantId, headers, vertxContext).compose(res -> {
     TenantLoading tl = new TenantLoading();
     // two sets of reference data files
     // resources ref-data/data1 and ref-data/data2 .. loaded to
     // okapi-url/instances and okapi-url/items respectively
+    Promise<Void> promise = Promise.promise();
     tl.withKey("loadReference").withLead("ref-data")
       .withIdContent()
       .add("data1", "instances")
       .add("data2", "items");
     tl.perform(tenantAttributes, headers, vertx, res1 -> {
       if (res1.failed()) {
-        handler.handle(Future.succeededFuture(PostTenantResponse
-          .respond500WithTextPlain(res1.cause().getMessage())));
+        promise.fail(res1.cause().getMessage());
         return;
       }
-      handler.handle(res);  // HTTP status: 200 for upgrade, 201 for install
+      promise.complete(); // OK
     });
-  }, context);
+    return promise.future();
```

Comment:
More concise when using Future.future(promise -> ...):
```
    return Future.future(promise -> tl.perform(tenantAttributes, headers, vertx, promise)
        .mapEmpty();
```

Reply:
Doesn't work for me.. Is this correct?

```
   Promise<Integer> promise = Promise.promise();
   tl.perform(attributes, headers, vertx, promise);
   return promise.future().mapEmpty();
```

How can that be turned into a one-liner?

Reply:
Yes, done.